### PR TITLE
click: 0.3.2 -> 0.4.2

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
fixing broken packages in rryantm logs

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @mbode

### CODE REVIEW

File: pkgs/applications/networking/cluster/click/default.nix [hunk 1]
```diff
@@ -4,18 +4,16 @@ with rustPlatform;
...
     rev = "v${version}";
     owner = "databricks";
     repo = "click";
-    sha256 = "0sbj41kypn637z1w115w2h5v6bxz3y6w5ikgpx3ihsh89lkc19d2";
+    sha256 = "18mpzvvww2g6y2d3m8wcfajzdshagihn59k03xvcknd5d8zxagl3";
   };
 
-  cargoSha256 = "1179a17lfr3001vp1a2adbkhdm9677n56af2c0zvkr18jas6b2w7";
-
-  patches = [ ./fix_cargo_lock_version.patch ];
```
Comment (worldofpeace) [inline pkgs/applications/networking/cluster/click/default.nix]: You need to delete the patch in the nixpkgs tree.

**Review by worldofpeace — CHANGES_REQUESTED** [review]

**Comment by worldofpeace** [comment]
Changelog: https://github.com/databricks/click/blob/v0.4.2/CHANGELOG.md

**Comment by jonringer** [comment]
@worldofpeace  Remedied the issues, and looks good to merge

```
4 built, 6 copied (12.0 MiB), 3.2 MiB DL]
https://github.com/NixOS/nixpkgs/pull/65857
1 package were build:
click
```

**Comment by mmahut** [comment]
@GrahamcOfBorg build click

**Review by mmahut — APPROVED** [review]
- [X] reviewed the diff and commit messages
- [x] made sure ofBorg build succeeded for all applicable platforms
- [X] run nix-review without any failures
- [X] run and tested the binaries

**Pull request merged**